modules/seqexec/engine/src/main/scala/seqexec/engine/package.scala

```diff
@@ -104,12 +105,19 @@ package object engine {
     Action(kind, Stream.eval(t), Action.State(Action.ActionState.Idle, Nil))
 
   /**
-    * An `Execution` is a group of `Action`s that need to be run in parallel
+    * `ParallelActions` is a group of `Action`s that need to be run in parallel
     * without interruption. A *sequential* `Execution` can be represented with
     * an `Execution` with a single `Action`.
     */
-  type Actions[F[_]] = List[Action[F]]
-
-  type Results[F[_]] = List[Result[F]]
+  type ParallelActions[F[_]] = NonEmptyList[Action[F]]
+
+  implicit class ListParallelActionsOps[F[_]](val v: List[Action[F]]) extends AnyVal {
+    def prepend(ac: List[ParallelActions[F]]): List[ParallelActions[F]] =
+      if (v.isEmpty) {
+        ac
+      } else {
+        NonEmptyList.fromListUnsafe(v) :: ac
+      }
```

Comment:
I think you can say `NonEmptyList.fromList(v).foldRight(ac)(_ :: _)` for the body here. If not you can at least do `NonEmptyList.fromList(v) match ...`